src/analysis/inner_product_space/spectrum.lean

```diff
@@ -130,8 +131,25 @@ show (⨆ μ : {μ // (eigenspace T μ) ≠ ⊥}, eigenspace T μ)ᗮ = ⊥,
 by rw [supr_ne_bot_subtype, hT.orthogonal_supr_eigenspaces_eq_bot]
 
 include dec_𝕜
+omit hT
+/-- The eigenspaces of a self-adjoint operator on a finite-dimensional inner product space `E` gives
+an internal direct sum decomposition of `E`. -/
+noncomputable instance direct_sum_decomposition [hT : fact T.is_symmetric] :
```

Comment:
Could we have a comment explaining why you are using `fact` here? (I think this is good practice, as it can be hard to decipher the reasons for this from the code.)

Reply:
Added.